src/main/scala/co/datamechanics/delight/DelightStreamingConnector.scala

```diff
@@ -287,13 +291,15 @@ class DelightStreamingConnector(sparkConf: SparkConf) extends Logging {
     *
     * - The threads are started only is they have not started yet.
     */
-  private def startIfNecessary(): Unit = {
+  private def startIfNecessary(): Unit = Utils.time(shouldLogDuration, "startIfNecessary") {
     if(started.compareAndSet(false, true)) {
       val pollingThread = new Thread {
         override def run() {
           while (true) {
+            val start = System.currentTimeMillis()
             publishPendingEvents()
-            Thread.sleep(currentPollingInterval.toMillis)
+            val end = System.currentTimeMillis()
+            Thread.sleep(math.max(currentPollingInterval.toMillis - (end - start), 0))
```

Comment:
Could we  mutualize the creation of a thread w/ sleep in an Utils method ?
Because we repeat almost the same code for the polling thread and the heartbeat thread

Reply:
Done in [`4936a15` (#9)](https://github.com/datamechanics/delight/pull/9/commits/4936a158611d5efd34f91744cd6e6383714b48a2)